MangaRipper.Core/Outputers/OutputFactory.cs

```diff
@@ -13,6 +13,8 @@ public IOutputer CreateOutput(OutputFormat format)
                     return new FolderOutput();
                 case OutputFormat.CBZ:
                     return new CbzOutput();
+                case OutputFormat.Counter: // no IOutputer implementation
```

Comment:
This design can be improved. For example, we should make it an option in FolderOuput or and Post-Output.

Reply:
Not sure what you mean by this.
The main point was to set the simple meaningful name instead of long ones before saving it on the FileSystem.

I set it as "OutputFormat" because it's the easiest way to access it without changing "DownloadChapterTask".

Could you please explain your suggestion it a little bit more.

Reply:
I think output format is folder or cbz but naming files should be a difference mechanism.

Currently, we download images into a temp folder and create 'output' folder or cbz based on images in temp folder. We can add a step to renaming files in the temp folder before create output. It's clear and less complex.

Later, we can extend it to more complex renaming scenario or provide many setting to rename files much easier.

Reply:
Currently, images are saved in the temp folder with their original names ("DownloadImages" method). And according to #133 it is the place where it failed. To prevent it from happening, I have put the renaming operation there.

You proposal require making the changes in "DownloadChapterTask" class and in the "DownloadImages" method. And if it's pretty clear what to do with "DownloadChapterTask" class, "DownloadImages" method is not so clear.
Should we simply check and cut the name if it has a long name, or should we save the files with random names and return a new class which will keep the temp names with original names (and maybe counter) and later uses them with "Output" methods?

Reply:
@NguyenDanPhuong  ping

Reply:
I don't think this is relevant now. I will close this. Thanks.